src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/upgrade/upgrade.component.html

```diff
@@ -54,34 +74,53 @@
           {{ (healthData.mgr_map | mgrSummary).total }}
         </span>
       </div>
-
-      <div class="d-flex mt-3 p-0">
-        <dl class="w-50"
-            *ngIf="fsid$ | async as fsid">
-          <dt class="bold mt-5"
-              i18n>Cluster FSID</dt>
-          <dd class="mt-2">{{ fsid }}</dd>
-          <dt class="bold mt-5"
-              i18n>Release Image</dt>
-          <dd class="mt-2">{{ upgradeInfo.image }}</dd>
-          <dt class="bold mt-5"
-              i18n>Registry</dt>
-          <dd class="mt-2">{{ upgradeInfo.registry }}</dd>
-        </dl>
+    </cd-card>
+    <div class="d-flex mt-3">
+      <ng-container *ngIf="{upgradeInfo: upgradeInfo$ | async, error: upgradeInfoError$ } as upgrade">
+        <ng-container *ngIf="!upgrade.upgradeInfoError && upgrade.upgradeInfo as upgradeInfo; else loadingDetails">
+          <dl class="w-50"
+              *ngIf="fsid$ | async as fsid">
+            <dt class="bold mt-5"
+                i18n>Cluster FSID</dt>
+            <dd class="mt-2">{{ fsid }}</dd>
+            <dt class="bold mt-5"
+                i18n>Release Image</dt>
+            <dd class="mt-2">{{ upgradeInfo.image }}</dd>
+            <dt class="bold mt-5"
+                i18n>Registry</dt>
+            <dd class="mt-2">{{ upgradeInfo.registry }}</dd>
+          </dl>
+        </ng-container>
+      </ng-container>
```

Comment:
Yeah, I followed the openshift designs which has it like this. There are room for improvements here though. For instance I'll need to provide the `upgrade history` some point later on when its implemented in cephadm. Once we add in those info probably this will be rectified. Meanwhile I put a video in the above comments by removing these details.